### DISCUSSION TITLE
Menu Items Not sticking

### DISCUSSION THREAD

**Discussion by cloudbr34k84** [elchininet/custom-sidebar#240] [General]
Hey, stumbled across from just now digging into theming etc. 
Im struggling to understand how to use this with my already reordered menu? 

I'm just playing around with the below
```
title: 'Home {{ as_timestamp(states("sensor.time")) | timestamp_custom("%H:%M") }}'
icon_color_selected: var(--accent-color)
order:
  - item: hacs
    order: 11
    notification: |
      [[[
        const outdatedHacsEntities = Object.values(entities.update).filter(
          (entity) => entity.platform === 'hacs' && is_state(entity.entity_id, 'on')
        );
        return outdatedHacsEntities.length || '';
      ]]]
  - new_item: true
    item: Automations
    href: "/config/automation"
    icon: mdi:robot
    info: |
      {{ states.automation | selectattr('state', 'eq', 'on') | list | count }} active
      {{ states.automation | selectattr('state', 'eq', 'off') | list | count }} inactive
    order: 12
```
![image](https://github.com/user-attachments/assets/ea0a0982-a700-4203-81ce-66fab0aaa88e)

1. It's not renaming my menu title
2. when I go into the menu edit mode by holding down the  menu name it then reorders to the below
![image](https://github.com/user-attachments/assets/63bec83d-eca7-47fb-933a-d532377e52c0)

I have cleared cached that many times. In pointers would be great!

**Comment by elchininet** [comment]
Hi @cloudbr34k84,
Please, open your browser console and check that you don't have any error/warning there. The plugin throws errors/warning when it finds something that is not right.

It is recommended that you follow this from the documentation:

>It is recommended to start with the simplest configuration and start to extend it step by step, in that way if you suffer any issue, it is easy to identify with which code block it started.

**Reply by elchininet** [reply]
One question, when you say "already reordered menu" do you mean that you changed the order using the Home Assistant native feature to do so?

**Reply by cloudbr34k84** [reply]
> One question, when you say "already reordered menu" do you mean that you changed the order using the Home Assistant native feature to do so?

Hey @elchininet thanks for replying. You I have used the native feature to remove and move the menu items around. 

I'll try again over the weekend and check for errors

**Reply by elchininet** [reply]
>I have used the native feature to remove and move the menu items around.

If you are going to use the plugin you need to set up the order with the plugin or the order will be messed up. There is even an option in the plugin to avoid that the native order gets used.

**Reply by cloudbr34k84** [reply]
Im not see any errors related to the component

yeah, this is where I had trouble understanding what I should do, whether to have a combination of both or just use this? 
If I want to just use this hacs component, do I need to remove all the side menu items?

**Reply by elchininet** [reply]
>Im not see any errors related to the component

Then it it weird that your title is not being applied. Did you check that the entity `sensor.time` exists? Also, did you test the template in the Developer tools, just to check that it renders correctly?

>yeah, this is where I had trouble understanding what I should do, whether to have a combination of both or just use this?
If I want to just use this hacs component, do I need to remove all the side menu items?

The Home Assistant functionality and this plugin should not be used together. If you are going to use the plugin stick to it and hide or change the order of the items only with the plugin, if you are going to use the Home Assistant functionality, then you should not use the plugin to change the order or to hide items. If you want to use the plugin for the order and the visibility of the items, then it is advisable that you start from scratch without any Home Assistant customisation. To avoid logging-out to remove the customisations, just remove these entries from Application > Local Sotorage in the Dev tools of your browser: `sidebarHiddenPanels`and `sidebarPanelOrder`.

**Reply by cloudbr34k84** [reply]
ok cheers, i think im getting the hang of it now, thank you

**Reply by elchininet** [reply]
What about the title?

**Reply by cloudbr34k84** [reply]
the title is there, its just if if you go to edit the side menu using the built in feature it reverts to original name of Home Assistant, but then I press F5 and its back

**Reply by elchininet** [reply]
> the title is there, its just if if you go to edit the side menu using the built in feature it reverts to original name of Home Assistant, but then I press F5 and its back
> 

Ah OK, I understood that it was not applied. But yes, every time that Home Assistant re-renders any element it loses the customisations. That is why if you plan to have those customisations done with the plugin, you should block that Home Assistant feature (you can even do it for others without logging-in with their accounts or their devices)

**Comment by fabriba** [comment]
Hi, does anyone knows if there is any way to combine that javascript with jinja2?

[[[
        const outdatedHacsEntities = Object.values(entities.update).filter(
          (entity) => entity.platform === 'hacs' && is_state(entity.entity_id, 'on')
        );
        return outdatedHacsEntities.length || '';
      ]]]
    
and

{% set totalUpdates = states.update | selectattr('state', 'eq', 'on') | list | length %}
      {{ totalUpdates > 0  }}

**Reply by elchininet** [reply]
Hi @fabriba,
Do you mean the code to get the `HACS` updates? If it is that, there is an example in the [Jinja Templates section](https://elchininet.github.io/custom-sidebar/templates/jinja-templates).
